crates/pgt_workspace/src/workspace/server.rs

```diff
@@ -155,31 +199,62 @@ impl WorkspaceServer {
 }
 
 impl Workspace for WorkspaceServer {
+    fn register_project_folder(
+        &self,
+        params: RegisterProjectFolderParams,
+    ) -> Result<ProjectKey, WorkspaceError> {
+        let current_project_path = self.get_current_project_path();
+        debug!(
+            "Compare the current project with the new one {:?} {:?} {:?}",
+            current_project_path,
+            params.path.as_ref(),
+            current_project_path.as_deref() != params.path.as_ref()
+        );
+
+        let is_new_path = match (current_project_path.as_deref(), params.path.as_ref()) {
+            (Some(current_project_path), Some(params_path)) => current_project_path != params_path,
+            _ => true,
```

Comment:
can we have a current_project_path but params_path? Then it wouldn't be a new path, right?

Reply:
yeah, added it. although I am not sure when this an happen :D